src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            FruitTransaction.Operation operation = getOperationByCode(parts[0]);
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
+        }
+        return transactions;
+    }
+
+    private FruitTransaction.Operation getOperationByCode(String code) {
+        return Arrays.stream(FruitTransaction.Operation.values())
+                .filter(op -> op.getCode().equals(code))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Unknown operation: " + code));
```

Comment:
Throwing an `IllegalArgumentException` is appropriate here, but make sure to handle this exception in the calling method to avoid crashing the application. You could catch the exception and rethrow it as a custom application-specific exception with a more informative message for the end-user.